### PR TITLE
New fixes

### PR DESCRIPTION
**Pull request by oleg-shumar**

### CODE REVIEW

File: inc/ajax.php [hunk 1]
```diff
@@ -18,7 +18,7 @@ function sell_media_add_to_cart() {
 	global $sm_cart;
 
 	if( !isset( $_POST['_wpnonce'] ) || isset( $_POST['_wpnonce'] ) && !wp_verify_nonce( $_POST['_wpnonce'], 'sell_media_add_cart_action') ) {
-		echo 0;
+		echo esc_html( '0' );
```
Comment (thadallender) [inline inc/ajax.php:21]: @oleg-shumar Should we be using `wp_die()` here to replace the `echo` and `exit`?

https://wordpress.stackexchange.com/questions/250280/best-way-to-end-wordpress-ajax-request-and-why

Comment (oleg-shumar) [reply inc/ajax.php:21]: I will comment this privately

**Comment by maharzan** [comment]
Issues I found:

- Collections: After you choose an image in icon and save it, the icon is still there. It should be blank once saved. 
- Licenses: Maybe we should put a default value in the Markup section. I am not sure if there was one before. perhaps 0%.
- Pricing: The settings seems to be saving now but I am getting these notices:

    Notice: Undefined variable: wp in /home3/maharzan/public_html/gpp/wp-content/plugins/sell-media/themes/price-listings-tabs-content.php on line 8

    Notice: Trying to get property 'request' of non-object in /home3/maharzan/public_html/gpp/wp-content/plugins/sell-media/themes/price-listings-tabs-content.php on line 8

    Warning: Cannot modify header information - headers already sent by (output started at /home3/maharzan/public_html/gpp/wp-admin/includes/template.php:2532) in /home3/maharzan/public_html/gpp/wp-includes/pluggable.php on line 1340

    Warning: Cannot modify header information - headers already sent by (output started at /home3/maharzan/public_html/gpp/wp-admin/includes/template.php:2532) in /home3/maharzan/public_html/gpp/wp-includes/pluggable.php on line 1343


- Pricing: Under Download Settings there is this option: High Resolution File Availability. If that is set to "Can be Purchased", then under Settings -> Pricing, Hide Original Price should be No and vice versa. I think currently its opposite.

Archive page: I still see issues on this page, not sure what it is.. wp_kses array? https://nhuja.com/gpp/shop/

**Comment by maharzan** [comment]
Ah the Json is actually coming from this script

    `<script type="application/ld+json">{"@context":"https:\/\/schema.org\/","@type":"ImageObject","contentUrl":"https:\/\/nhuja.com\/gpp\/wp-content\/uploads\/2022\/02\/johannes-plenio-qkfxBc2NQ18-unsplash-600x600.jpg","license":"https:\/\/nhuja.com\/gpp\/shop\/new-test-2\/johannes-plenio-qkfxbc2nq18-unsplash\/","acquireLicensePage":"https:\/\/nhuja.com\/gpp\/shop\/new-test-2\/johannes-plenio-qkfxbc2nq18-unsplash\/"}</script>`

Can we remove this?

**Comment by maharzan** [comment]
The pagination is also showing HTML

This line https://github.com/oleg-shumar/Sell-Media/blob/new_fixes/themes/archive.php#L180

        if ( ! $children ) {
				echo esc_html( sell_media_pagination_filter( $wp_query->max_num_pages ) );
			}

should be
			if ( ! $children ) {
				echo wp_kses( sell_media_pagination_filter( $wp_query->max_num_pages ), [
					'div' => [
						'id' => true,
						'class' => true,
						'data-*' => true,
					],
					'span' => [
						'id' => true,
						'class' => true,
						'aria-*' => true,
					],
					'a' => [
						'href' => true,
						'id' => true,
						'target' => true,
						'class' => true,
						'data-*' => true,
					],
				]);
			}

**Comment by oleg-shumar** [comment]
> Ah the Json is actually coming from this script
> 
> ```
> `<script type="application/ld+json">{"@context":"https:\/\/schema.org\/","@type":"ImageObject","contentUrl":"https:\/\/nhuja.com\/gpp\/wp-content\/uploads\/2022\/02\/johannes-plenio-qkfxBc2NQ18-unsplash-600x600.jpg","license":"https:\/\/nhuja.com\/gpp\/shop\/new-test-2\/johannes-plenio-qkfxbc2nq18-unsplash\/","acquireLicensePage":"https:\/\/nhuja.com\/gpp\/shop\/new-test-2\/johannes-plenio-qkfxbc2nq18-unsplash\/"}</script>`
> ```
> 
> Can we remove this?
> 
> It looks like its coming from this line: https://github.com/oleg-shumar/Sell-Media/blob/new_fixes/inc/class-layouts.php#L428
> 
> It might be used by licensing. If not, maybe we could just do div instead of script as I think wp_kses doesn't allow script.

**About this - it was already updated a week ago** with 
'script' => array('type' => true), in template-tags.php
and on my dev display issue was fixed with that. Are you sure you see it in latest version?
> 

Thanks for noticing that

> Collections: After you choose an image in icon and save it, the icon is still there. It should be blank once saved.

But why? Isn't hiding an image makes confusion to people?

>Pricing: Under Download Settings there is this option: High Resolution File Availability. If that is set to "Can be Purchased", >then under Settings -> Pricing, Hide Original Price should be No and vice versa. I think currently its opposite.

That was not ever changed by me, as it is out of escaping scope. If it's a quick thing, I'll do it BTW

**Comment by oleg-shumar** [comment]
Fixed pricing
Added default markup to field

**Comment by oleg-shumar** [comment]
Please test new changes

**Comment by maharzan** [comment]
The script is also loading in themes/archive.php, thats where the code is.

**Comment by maharzan** [comment]
I am not sure about the pricing page. I think we could just replicate the same parameters as to having confusing options. Lets just have the same options as in the settings-pricing page.

**Comment by oleg-shumar** [comment]
> The script is also loading in themes/archive.php, thats where the code is coming from on that shop page.

Pushed, updated wp_kses there

**Comment by maharzan** [comment]
- Collection: Icon still not removed after upload/save.
- Licenses: The default markup could add % so its 0% (not big deal) but after saving, the default goes blank and if you save without changing markup value, it is saving only %
test
.
- Pricing menu: I think what I said earlier should be good.

The option: **High Resolution File Availability** when set to **Cannot be purchased**, its listing the Original File/Price which should be otherwise. So, please set that to **Can be purchased** if Settings-> Pricing **Hide Original Price** is set to **No** and vice versa.

**Comment by maharzan** [comment]
test
The json is also coming in quick view.

**Comment by maharzan** [comment]
Also, I am testing Lightbox, so when you add an image to lightbox by clicking "Save to Lightbox" near the Add the Cart button. It seems to save but When I go to lightbox page, its not there. After checking this line

https://github.com/oleg-shumar/Sell-Media/blob/new_fixes/inc/lightbox.php#L136

is not echoing any value. Looks like sanitize_key is removing all the json style value.

**Comment by oleg-shumar** [comment]
> Collection: Icon still not removed after upload/save.
I don't agree with it, but returned. Now it's being removed after uploading and clicking save. Old code returned back, one line.

**Comment by oleg-shumar** [comment]
>Licenses: The default markup could add % so its 0% (not big deal) but after saving, the default goes blank and if you save without changing markup value, it is saving only %

Updated

**Pull request merged**